src/atom/mod.rs

```diff
@@ -171,6 +178,34 @@ impl<'a> From<&'a str> for Atom {
     }
 }
 
+impl<'a> From<&'a str> for Atom {
+    #[inline]
+    fn from(string_to_add: &str) -> Atom {
+        Atom::from(Cow::Borrowed(string_to_add))
+    }
+}
+
+impl From<String> for Atom {
+    #[inline]
+    fn from(string_to_add: String) -> Atom {
+        Atom::from(Cow::Owned(string_to_add))
+    }
+}
+
+impl From<Atom> for String {
```

Comment:
This is only a win over using `str::to_owned` through `Deref<Target=str>` when the refcount happens to be 1, which seems seriously improbable. This this adds unsafe code, I’m mildly opposed to adding it without demonstrated perf wins.